dbt/adapters/spark/impl.py

```diff
@@ -108,53 +115,64 @@ def drop_relation(self, relation, model_name=None):
+        rows = [
+            row for row in raw_rows[0:pos]
+            if not row['col_name'].startswith('#')
+        ]
+        metadata = {
+            col['col_name']: col['data_type'] for col in raw_rows[pos + 1:]
+        }
 
-        column_names = (
-            'table_database',
-            'table_schema',
-            'table_name',
-            'table_type',
-            'table_comment',
-            'table_owner',
-            'column_name',
-            'column_index',
-            'column_type',
-            'column_comment',
+        return [SparkColumn(
+            relation.database,
+            relation.schema,
+            relation.name,
+            relation.type,
+            metadata.get(KEY_TABLE_OWNER),
+            column['col_name'],
+            idx,
+            column['data_type']
+        ) for idx, column in enumerate(rows)]
+
+    def get_properties(self, relation: Relation) -> Dict[str, str]:
+        properties = self.execute_macro(
+            FETCH_TBLPROPERTIES_MACRO_NAME,
+            kwargs={'relation': relation}
         )
+        return {key: value for (key, value) in properties}
+
+    def get_catalog(self, manifest: Manifest) -> agate.Table:
+        schemas = manifest.get_used_schemas()
+
+        def to_dict(d: any) -> Dict:
+            return d.__dict__
 
         columns = []
         for (database_name, schema_name) in schemas:
             relations = self.list_relations(database_name, schema_name)
             for relation in relations:
-                logger.debug("Getting table schema for relation {}".format(relation))  # noqa
-                table_columns = self.get_columns_in_relation(relation)
-                rel_type = self.get_relation_type(relation)
-
-                for column_index, column in enumerate(table_columns):
-                    # Fixes for pseudocolumns with no type
-                    if column.name in (
-                        '# Partition Information',
-                        '# col_name'
-                    ):
-                        continue
-                    elif column.dtype is None:
-                        continue
-
-                    column_data = (
-                        relation.database,
-                        relation.schema,
-                        relation.name,
-                        rel_type,
-                        None,
-                        None,
-                        column.name,
-                        column_index,
-                        column.data_type,
-                        None,
-                    )
-                    column_dict = dict(zip(column_names, column_data))
-                    columns.append(column_dict)
-
-        return dbt.clients.agate_helper.table_from_data(columns, column_names)
+                logger.debug("Getting table schema for relation {}", relation)
+                columns += list(map(to_dict, self.get_columns_in_relation(relation)))
```

Comment:
There's only one `flake8` error, and it's on this line:
```
dbt/adapters/spark/impl.py:176:80: E501 line too long (85 > 79 characters)
```

Reply:
Whoops, thanks!